matlab/tools/packageMatlabInterface.m

```diff
@@ -0,0 +1,58 @@
+% Licensed to the Apache Software Foundation (ASF) under one
+% or more contributor license agreements.  See the NOTICE file
+% distributed with this work for additional information
+% regarding copyright ownership.  The ASF licenses this file
+% to you under the Apache License, Version 2.0 (the
+% "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing,
+% software distributed under the License is distributed on an
+% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+% KIND, either express or implied.  See the License for the
+% specific language governing permissions and limitations
+% under the License.
+
+toolboxFolder = string(getenv("ARROW_MATLAB_TOOLBOX_FOLDER"));
+outputFolder = string(getenv("ARROW_MATLAB_TOOLBOX_OUTPUT_FOLDER"));
+toolboxVersionRaw = string(getenv("ARROW_MATLAB_TOOLBOX_VERSION"));
+
+% Output folder must exist.
+mkdir(outputFolder);
+
+disp("Toolbox Folder: " + toolboxFolder);
+disp("Output Folder: " + outputFolder);
+disp("Toolbox Version Raw: " + toolboxVersionRaw);
+
+% Note: This string processing heuristic may not be robust to future
+% changes in the Arrow versioning scheme.
+dotIdx = strfind(toolboxVersionRaw, ".");
+numDots = numel(dotIdx);
+if numDots >= 3
+    toolboxVersion = extractBefore(toolboxVersionRaw, dotIdx(3));
+else
+    toolboxVersion = toolboxVersionRaw;
+end
+
+disp("Toolbox Version:" + toolboxVersion);
+
+identifier = "ad1d0fe6-22d1-4969-9e6f-0ab5d0f12ce3";
+opts = matlab.addons.toolbox.ToolboxOptions(toolboxFolder, identifier);
```

Comment:
Hi @kou, 

I'll update the script to set `AuthorName` to `Apache Arrow` and `AuthorEmail` to `dev@arrow.apache.org`. 

Regarding including the license, the following repository lists best practices for MATLAB Toolboxes: [mathworks/toolboxdesign](https://github.com/mathworks/toolboxdesign). The recommendation is to put a `License.txt` file in the root directory of the toolbox. Should we include the top-level [`LICENSE.txt`](https://github.com/apache/arrow/blob/main/LICENSE.txt) and [`NOTICE.txt`](https://github.com/apache/arrow/blob/main/NOTICE.txt) files within the MLTBX file? We looked at the Python wheel(s) files for reference, and they seem to package both files.

Best,
Sarah

Reply:
Yes. We must add `LICENSE.txt` and `NOTICE.txt` into the MLTBX file.

See also:
* The ASF release policy:
  * Artifacts https://www.apache.org/legal/release-policy.html#artifacts (The MLTBX file is a compiled package in this document)
  * Licencing https://www.apache.org/legal/release-policy.html#licensing
  * Licensing Documentation https://www.apache.org/legal/release-policy.html#licensing-documentation
* Assembling LICENSE and NOTICE files https://infra.apache.org/licensing-howto.html 
  * The MLTBX file is a binary distribution ihttps://infra.apache.org/licensing-howto.html#binary n this document
  * I think that the MLTBX file includes (bundles) at least https://github.com/mathworks/libmexclass . We need add license information for it. https://infra.apache.org/licensing-howto.html#bundled-vs-non-bundled

Reply:
Hi @kou,

Just wanted to provide a quick status update on this. We've been working on trying to understand what we need to include in LICENSE.txt/NOTICE.txt from a MEX perspective. We know we need to include a pointer about libmexclass, but we're still evaluating if there is more we should be including relating to MEX build artifacts. 

Note, we are approaching the holiday season in the US, so there may be a bit of a delay in moving this PR forward. Apologies for any inconvenience this delay causes. Thanks for helping us move this PR forward!

Reply:
Quick Update:

We're still consulting with other MathWorkers about the re-distribution of MEX binaries in order to ensure we are doing the right thing with respect to the ASF's licensing policies. Unfortunately, this is taking longer than we expected, but we will update this PR as soon as we can. We really appreciate everyone's support in helping us figure this out.

Reply:
Hi Everyone,
 
Thanks very much for patiently waiting on us to move this forward.
 
We realized that we need to ensure that the licenses of any MEX artifacts bundled with the released MLTBX file must be compatible with the ASF 3rd Party License Policy.
 
After several rounds of discussion with some colleagues at MathWorks (unfortunately delayed due to the holiday season in the US), we came up with a license ([attached to this comment](https://github.com/apache/arrow/files/13955180/license.txt)) that we hope will allow us to ensure that we are abiding by the ASF's 3rd Party License Policy.
 
The attached license is a modified version of the standard MathWorks license that applies to MEX files. The intent of these modifications is to meet the requirements for inclusion as a ["Category B"](https://www.apache.org/legal/resolved.html#category-b) license according to the ASF 3rd Party License Policy.
 
Our goal is to make sure we are doing the right thing here, and we understand this may need further input from ASF Legal and the rest of the community. Please let us know what we can do to help move this forward. We sincerely appreciate everyone's support as we navigate these licensing requirements.
 
**Link to proposed license:** [license.txt](https://github.com/apache/arrow/files/13955180/license.txt)

Best,
 
Sarah and Kevin

Reply:
Thanks for sharing this!

Let's discuss this on `dev@arrow.apache.org` to get more attention. We should think about license carefully.

Reply:
I agree! I'll start a discussion on the mailing list.

Reply:
After reviewing the relevant sections from https://www.apache.org/legal/release-policy.html and https://infra.apache.org/licensing-howto.html, we i[dated the `LICENSE.txt` and `NOTICE.txt` files as detailed below:

1. Added a "pointer" to `mathworks/libmexclass` at the bottom of `LICENSE.txt`.
2. Added a note about the fact that the project includes software from The MathWorks, Inc. (i.e. all of the contents of the distributed MLTBX files, other than mathworks/libmexclass and libarrow) which MathWorks is granting the Apache Software Foundation the right to distribute under the standard Apache Version 2.0 license.

We believe these updates should be sufficient to address the bundling of the `mathworks/libmexclass` binaries and all other contents in the distributed MLTBX files. However, please let us know if there is anything else we should be including in the `LICENSE.txt` or `NOTICE.txt` files.

Thanks again for everyone's help!